tests/puppeteer/campaigns/functional/BO/customers/customers/01_filter&QuickEditCustomers.js

```diff
@@ -0,0 +1,313 @@
+        'select',
+        'newsletter',
+        DefaultAccount.newsletter,
+      );
+      const numberOfCustomersAfterFilter = await this.pageObjects.customersPage.getNumberFromText(
+        this.pageObjects.customersPage.customerGridTitle);
+      await expect(numberOfCustomersAfterFilter).to.be.at.most(numberOfCustomers);
+      /* eslint-disable no-await-in-loop */
+      for (let i = 1; i <= numberOfCustomersAfterFilter; i++) {
+        const textColumn = await this.pageObjects.customersPage.getTextContent(
+          this.pageObjects.customersPage.customersListTableColumn.replace('%ROW', i).replace('%COLUMN', 'newsletter'),
+        );
+        await expect(textColumn).to.contains('check');
+      }
+      /* eslint-enable no-await-in-loop */
+    });
+    it('should reset all filters', async function () {
+      await this.pageObjects.customersPage.resetFilter();
+      const numberOfCustomersAfterReset = await this.pageObjects.customersPage.getNumberFromText(
+        this.pageObjects.customersPage.customerGridTitle);
+      await expect(numberOfCustomersAfterReset).to.equal(numberOfCustomers);
+    });
+    it('should filter by Partner Offers \'YES\'', async function () {
+      await this.pageObjects.customersPage.filterCustomers(
+        'select',
+        'optin',
+        DefaultAccount.partnerOffers,
+      );
+      const numberOfCustomersAfterFilter = await this.pageObjects.customersPage.getNumberFromText(
+        this.pageObjects.customersPage.customerGridTitle);
+      await expect(numberOfCustomersAfterFilter).to.be.at.most(numberOfCustomers);
+      /* eslint-disable no-await-in-loop */
+      for (let i = 1; i <= numberOfCustomersAfterFilter; i++) {
+        const textColumn = await this.pageObjects.customersPage.getTextContent(
+          this.pageObjects.customersPage.customersListTableColumn.replace('%ROW', i).replace('%COLUMN', 'optin'),
+        );
+        await expect(textColumn).to.contains('check');
+      }
+      /* eslint-enable no-await-in-loop */
+    });
+    it('should reset all filters', async function () {
+      await this.pageObjects.customersPage.resetFilter();
+      const numberOfCustomersAfterReset = await this.pageObjects.customersPage.getNumberFromText(
+        this.pageObjects.customersPage.customerGridTitle);
+      await expect(numberOfCustomersAfterReset).to.equal(numberOfCustomers);
+    });
+  });
+
+  describe('Quick Edit Customers', async () => {
+    // Steps
+    it('should filter by Email \'pub@prestashop.com\'', async function () {
+      await this.pageObjects.customersPage.filterCustomers(
+        'input',
+        'email',
+        DefaultAccount.email,
+      );
+      const numberOfCustomersAfterFilter = await this.pageObjects.customersPage.getNumberFromText(
+        this.pageObjects.customersPage.customerGridTitle);
+      await expect(numberOfCustomersAfterFilter).to.be.at.above(0);
+    });
+    it('should disable first Customer', async function () {
```

Comment:
It's maybe important to check the validation message after update, what do you think ?

Reply:
Nice catch but I have one problem with that : 
In the function `updateToggleColumnValue`, I check if the column has the value wanted before changing it.
SO, in the case of `actualValue != wantedValue`, scipt will change value and we will have the successful update message
but in the case of `actualValue == wantedValue`, script will not click to change value so **we won't have** the successful update message (and test will fail)
That why I choose to not check message, instead of that I check the real value of the column.

Reply:
In this case we can test the validation message only when the `actualValue != wantedValue`

Reply:
check done.
thanks @nesrineabdmouleh